### ISSUE TITLE
[Suggestion] Particle Emitter: RGB sliders change to HSV sliders

### ISSUE THREAD

**Issue by tlitookilakin** [brandon3055/Draconic-Evolution#1227]
Would it be possible to swap the RGB color range sliders on the **particle emitter** for HSV ones? I'm aware that the particle emitter is probably a low-priority block, but I thought I'd put the suggestion out there anyways.

The current setup, using RGB ranges, has a number of limitations, which mostly boil down to specificity. For example, with RGB sliders it's not possible to guarantee grey-scale while also having varying value, or to guarantee full saturation while having a broad range of colors, and achieving a specific range of shades can be very tricky. With HSV sliders, all of these things would be possible, and it would be much easier to achieve the desired result.

The main potential downsides I see for this implementation would be dev time required & the slight increase in CPU time when particles are generated, since the block would have to select a random HSV value and then convert it when the particle is generated. Additional CPU time would probably be pretty minimal, but it's still worth considering.

In short: 
Pro: more flexible, more specific
Con: programming time, performance loss

Again, I'm not expecting anything, this was mostly just a thought I had while I was tinkering with some of the particle systems in my base. Thank you for your time!
